galpy/orbit_src/Orbit.py

```diff
@@ -3774,6 +3779,140 @@ def animate(self,*args,**kwargs): #pragma: no cover
         """
         return self._orb.animate(*args,**kwargs)
 
+    @classmethod
+    def from_name(cls, name, gaiadr2=False, searchr=None, vo=None, ro=None,
+                  zo=None, solarmotion=None):
+        """
+        NAME:
+
+            from_name
+
+        PURPOSE:
+
+            given the name of an object, retrieve coordinate information from an online catalogue for that object and return a corresponding orbit
+
+        INPUT:
+
+            name - the name of the object
+
+            gaiadr2= if True, will attempt to find the object in the Gaia DR2 catalogue by cross-matching with SIMBAD; otherwise will directly use the coordinates from SIMBAD (default=False)
+
+            searchr= angular radius used to cone search the Gaia archive for the desired object (arcsec; can be Quantity; default = 1 arcsec)
+
+            +standard Orbit initialization keywords:
+
+                ro= distance from vantage point to GC (kpc; can be Quantity)
+
+                vo= circular velocity at ro (km/s; can be Quantity)
+
+                zo= offset toward the NGP of the Sun wrt the plane (kpc; can be Quantity; default = 25 pc)
+
+                solarmotion= 'hogg' or 'dehnen', or 'schoenrich', or value in [-U,V,W]; can be Quantity
+
+        OUTPUT:
+
+            orbit containing the phase space coordinates of the named object
+
+        HISTORY:
+
+            2018-07-09 - Written - Mathew Bub (UofT)
+
+        """
+        if not _APY_LOADED:
+            raise ImportError(('astropy needs to be installed to use '
+                               'Orbit.from_name'))
+        if not _ASTROQUERY_LOADED:
+            raise ImportError(('astroquery needs to be installed to use '
+                               'Orbit.from_name'))
+
+        # query SIMBAD for the named object
+        simbad= Simbad()
+        simbad.add_votable_fields('ra(d)', 'dec(d)', 'plx', 'pmra', 'pmdec',
+                                  'rv_value')
+        simbad.remove_votable_fields('main_id', 'coordinates')
+        try:
+            simbad_table= simbad.query_object(name)
+        except OSError: # pragma: no cover
+            raise ConnectionError('failed to connect to SIMBAD')
+        if not simbad_table:
+            raise ValueError('failed to find {} in SIMBAD'.format(name))
+        simbad_table= simbad_table.as_array().view((nu.float64,6))
+
+        # try to cross-match the SIMBAD coordinates with Gaia DR2
+        if gaiadr2:
+            from astroquery.gaia import Gaia
+            if searchr is None:
+                searchr= 1
+            if isinstance(searchr, units.Quantity):
+                searchr= searchr.to(units.deg).value
```

Comment:
This line is currently not hit by any of the tests, please add or edit one of the tests to give ``r`` with units so this gets hit.